content/terraform/v1.12.x/docs/language/block/provider.mdx

```diff
@@ -0,0 +1,419 @@
+terraform {
+  required_providers {
+    aws = {
+      source = "hashicorp/aws"
+    }
+  }
+}
+
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+
+  tags = {
+    Name = "Main VPC"
+  }
+}
+```
+
+</CodeBlockConfig>
+
+The `aws_vpc` resource inherits the same AWS provider configuration from the root module, and Terraform creates the `aws_vpc` resource in the `us-west-2` region. Child modules do not inherit provider source or version requirements, so you must explicitly define those within a child module. Learn more about [inheriting providers in modules](/terraform/language/modules/develop/providers#implicit-provider-inheritance).
+
+### Using an alternate provider configuration in a child module
+
+To use an aliased provider configurations in a child module, the child module must declare the alias using the `configuration_aliases` argument in the `required_providers` block.
+
+In the following example, the root module passes the `aws.west` alias to the `web-server` child module:
+
+<CodeBlockConfig highlight="13" filename="main.tf">
+
+```hcl
+provider "aws" {
+  region = "us-east-1"
+}
+
+provider "aws" {
+  alias  = "west"
+  region = "us-west-2"
+}
+
+module "web-server" {
+  source = "./modules/web-server"
+  providers = {
+    aws.west = aws.west
+  }
+}
+```
+
+</CodeBlockConfig>
+
+The following `configuration_aliases` argument declares that the child module expects to receive a provider configuration it can reference as `aws.west`. Without this declaration, Terraform raises an error when the module tries to reference `aws.west` in its resources.
+
+<CodeBlockConfig highlight="6, 12" filename="modules/web-server/main.tf">
+
+```hcl
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.0"
+      configuration_aliases = [aws.west]
+    }
+  }
+}
+
+data "aws_ami" "amazon_linux" {
+  provider = aws.west
+
+  #...
+}
+```
+
+</CodeBlockConfig>
+
+Modules have special requirements for providers, refer to [Providers within modules](/language/modules/develop/providers) to learn more.
+
+### Provider configuration with expressions
+
+In the following example, the provider configuration uses input variables and local values to set its arguments:
+
+<CodeBlockConfig highlight="15,18">
+
+```hcl
+variable "aws_region" {
+  description = "The AWS region to deploy resources in"
+  type        = string
+  default     = "us-west-2"
+}
+
+locals {
+  common_tags = {
+    Environment = "production"
+    Project     = "web-app"
+  }
+}
+
+provider "aws" {
+  region = var.aws_region
+
+  default_tags {
+    tags = local.common_tags
+  }
+}
+```
+
+</CodeBlockConfig>
+
+### Provider configuration without arguments
+
+In the following example, the provider block defines an empty configuration for the `random` provider. Terraform also assumes an empty default configuration for any provider that you do not explicitly configure with a `provider` block:
+
+```hcl
+provider "random" {
+
+}
+```
```

Comment:
```suggestion
```hcl
provider "random" { }
```
```

I see this style in examples more often than the line-break style for "empty" provider configuration.